### PR TITLE
Add sample buttons as input option

### PR DESCRIPTION
**Pull request by fhildeb**
### Feature

This PR adds sample buttons for assets and UP to the following pages:

- Data Fetcher
- Inspector

It is done using a `SampleInputAddress` component that calls the address updates on the related input fields and individual functions on both pages.

### Screenshot
Bildschirmfoto 2023-11-14 um 15 02 21

### CODE REVIEW

File: components/SampleAddressInput/SampleAddressInput.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { NetworkContext } from '../../contexts/NetworksContext';
+
+enum AddressType {
+  UP = 'UP',
+  Asset = 'Asset',
+}
+
+interface Props {
+  inputId: string;
+  onChangeFunction: (e: React.ChangeEvent<HTMLInputElement>) => void;
```
Comment (Esanim) [inline components/SampleAddressInput/SampleAddressInput.tsx:11]: you can keep it as is (name), but you might as well shorten to just onChange

**Review by Esanim — APPROVED** [review]

File: components/SampleAddressInput/SampleAddressInput.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        address = '0x9139def55c73c12bcda9c44f12326686e3948634';
+      } else if (network.name === 'TESTNET') {
+        address = '0x9139def55c73c12bcda9c44f12326686e3948634';
+      }
+    } else if (type === AddressType.Asset) {
+      if (network.name === 'MAINNET') {
+        // TODO: Update to Mainnet Asset
+        address = '0x6395b330F063F96579aA8F7b59f2584fb9b6c3a5';
+      } else if (network.name === 'TESTNET') {
+        address = '0x6395b330F063F96579aA8F7b59f2584fb9b6c3a5';
```
Comment (Hugoo) [inline components/SampleAddressInput/SampleAddressInput.tsx:35]: Could you define these addresses in a const or global files - or at the very list at the top of this file.

It is important to give names to addresses as over time we don't know what is what.

```js
const SAMPLE_MAINNET_UP_ADDRESS = '';
const SAMPLE TESTNET_UP_ADDRESS = '0x9139def55c73c12bcda9c44f12326686e3948634';

// NOTE: what kind of asset btw?
const SAMPLE_MAINNET_ASSET = '';
const SAMPLE_TESTNET_ASSET = '0x6395b330F063F96579aA8F7b59f2584fb9b6c3a5';
```

Also, you are using the same addresses for testnet and mainnet - is it intended? Let's not hide little "bombs" in the code, just change the address to empty string for MAINNET. At least it is clear that it is missing.

Having a testnet address like it is right not is wrong and might lead to problems.

File: components/SampleAddressInput/SampleAddressInput.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+interface Props {
+  inputId: string;
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+const SampleAddressInput: React.FC<Props> = ({ inputId, onChange }) => {
+  const { network } = useContext(NetworkContext);
+
+  const changeInputAddress = (type: AddressType) => {
+    let address;
```
Comment (Hugoo) [inline components/SampleAddressInput/SampleAddressInput.tsx:22]:
```suggestion
    let address: string;
```

File: components/SampleAddressInput/SampleAddressInput.tsx (hunk 2 above)
Comment (Hugoo) [reply components/SampleAddressInput/SampleAddressInput.tsx:35]: Consider using a switch case

File: components/SampleAddressInput/SampleAddressInput.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      if (network.name === 'MAINNET') {
+        // TODO: Update to Mainnet Asset
+        address = '0x6395b330F063F96579aA8F7b59f2584fb9b6c3a5';
+      } else if (network.name === 'TESTNET') {
+        address = '0x6395b330F063F96579aA8F7b59f2584fb9b6c3a5';
+      }
+    }
+
+    if (address) {
+      const inputElement = document.getElementById(inputId) as HTMLInputElement;
```
Comment (Hugoo) [inline components/SampleAddressInput/SampleAddressInput.tsx:40]: we are working with react here - there should be no need to use plain js like that:

`document.getElementById(inputId) as HTMLInputElement;`

Consider using react with: refs:

https://react.dev/learn/manipulating-the-dom-with-refs

File: components/SampleAddressInput/SampleAddressInput.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      if (inputElement) {
+        inputElement.value = address;
+        onChange({
+          target: { value: address },
+        } as React.ChangeEvent<HTMLInputElement>);
+      }
+    }
```
Comment (Hugoo) [inline components/SampleAddressInput/SampleAddressInput.tsx:47]: no need for that - this looks weird

Comment (Hugoo) [reply components/SampleAddressInput/SampleAddressInput.tsx:47]: Where is the element with `{inputId}` class defined??

Let's try to not hijack the system by targetting dom elements like that. I'm sure there is another easiest way.

**Review by Hugoo — CHANGES_REQUESTED** [review]

**Comment by fhildeb** [comment]
I've applied the suggestions, however one element had to stay:

```js
    if (address && inputRef.current) {
      inputRef.current.value = address;
      onChange({
        target: { value: address },
      } as React.ChangeEvent<HTMLInputElement>);
    }
```

As we are having one component that is used on 2 pages with different `onChange` behavior/functions for `<input>` elements, we have to pass the function into the component, so it can be called.

`onChange` is only triggered from manual user inputs, not if done by code. Thats why the function must be triggered manually after.

**Comment by Hugoo** [comment]
@fhildeb i don't understand your comment. Please check my last commit: https://github.com/lukso-network/tools-erc725-inspect/pull/63/commits/b7b4b4f90d033ff3490ac3704e6db240ff7ed34e

**Review by Hugoo — APPROVED** [review]

**Pull request merged**